### ISSUE TITLE
Learning rate reduction not functional

### ISSUE THREAD

**Issue by tboo** [ludwig-ai/ludwig#3049]
**Describe the bug**
Despite the respective settings in the `trainer` section, the learning rate does not decay when a plateau is reached.

**To Reproduce**

I am using ludwig for image classification with the following .yaml
```
input_features:
- name: image_path
  type: image
  encoder: resnet
  resnet_size: 101
  num_filters: 16
  kernel_size: 3
  conv_stride: 1
  batch_norm_momentum: 0.9
  batch_norm_epsilon: 0.001
  num_fc_layers: 1
  output_size: 256
  use_bias: true
  bias_initializer: zeros
  norm: null
  norm_params: null
  activation: relu
  dropout: 0

output_features:
 - name: class
   type: category

trainer:
  epochs: 50
  batch_size: 16
  early_stop: 8
  learning_rate: 0.001
  reduce_learning_rate_on_plateau: 4
  reduce_learning_rate_on_plateau_patience: 4
  reduce_learning_rate_on_plateau_rate: 0.5
```
I train the model distributed with horovod in a ludwig stock docker container:
```
horovodrun -np 3 ludwig train --config model_definition.yaml --dataset annotations.csv --experiment_name resnet101
```

In this particular case the learning rate the model describes is even `0.003`. An Excerpt from `training_progress.json`:

```
    "best_increase_batch_size_eval_metric": Infinity,
    "checkpoint_number": 34,
    "epoch": 34,
    "last_improvement_steps": 6512,
    "last_increase_batch_size": 0,
    "last_increase_batch_size_eval_metric_improvement": 0,
    "last_increase_batch_size_steps": 0,
    "last_learning_rate_reduction": 0,
    "last_learning_rate_reduction_steps": 0,
    "learning_rate": 0.003,
    "num_increases_batch_size": 0,
    "num_reductions_learning_rate": 0,
    "steps": 27676,
```

**Expected behavior**
The learning rate should be reduced after 4 epochs on no further loss improvement/

**Environment (please complete the following information):**

- OS: Linux Docker environment
- Version: Docker ludwigai/ludwig-ray-gpu:master
- Python version 3.8.5
- Ludwig version 0.7.beta

**Additional context**
this does not seem to be ludwig version specific. I experienced a similar behavior with ludwig 0.5.5

Thanks so much for some feedback and keep up the great work.

**Comment by tgaddair** [comment]
Hey @tboo, thanks for reporting this issue. We recently made some big changes to learning rate reduction, so definitely want to make sure it's working correctly! 

I was able to repro the issue you're seeing in your environment and will take some time to dig into it tomorrow.

**Comment by tgaddair** [comment]
Hey @tboo, I found the issue which was due to a weird interaction with using multiple learning rate schedulers. I put together #3056 to address. Let me know if it resolves the issue for you!

**Comment by tboo** [comment]
Hey @tgaddair - Unfortunately the reduction still does not work for me with the latest ludwigai/ludwig-ray-gpu:master container. Meaning the early stop kicks in before any reduction of the learning rate. Which should not be the case with the above *.yaml file, when I understand it correctly. I wanted to test it with MNIST but `ludwig datasets download mnist` unfortunately doe not produce the `mnist_dataset.csv`, only the raw images and `mnist.parquet`.

**Comment by tgaddair** [comment]
Hey @tboo, sorry it didn't address your issue! Let me know if you have a standalone repro script I can run. 

One thing that might be causing a difference behavior here, though: early stopping is measured on to the VALIDATION split metrics by default, while learning rate reduction is measured on the TRAINING split loss metric by default. It would make sense that training metrics might still be improving while validation metrics have not (if the model is overfitting).

If you set:

```
trainer:
    reduce_eval_metric: accuracy
    reduce_eval_split: validation
```

Then it should be measuring against the same metric as early stopping. Can you try that and let me know how it goes?

cc @w4nderlust @justinxzhao we should consider changing the reduce LR on plateau defaults to be consistent with early stopping. What do you think?

Regarding MNIST, am I understanding correctly that you prefer to have the file as a CSV rather than a parquet? Any particular reason for that? It would be pretty easy to change it if it's more convenient to work with.

**Comment by w4nderlust** [comment]
The learning rate reduction on plateau should be on the training loss by default.
It's ment to address the cases where the training process lands in a valley of the loss landscape that is narrow and the current learning rate leads to bouncing off surfaces without being able to go further down in the valley because the gradient is updated of a step that is too large. Reducing the LR in these situations (which are all about the training loss landscape, not about the validation loss or accuracy) makes it possible to go further down the valley.
The effect should be like this: 
![image](https://user-images.githubusercontent.com/349256/218275474-0f5dc405-e9e6-41bb-bdf3-9460539baac0.png)

That said, obviously the mechanism should work, and if after 4 epochs of not decrease in training loss it does not kick in, we need to fix it ;)

**Comment by tgaddair** [comment]
Thanks @w4nderlust, sounds like we're happy with the behavior then, it's just a matter of setting the right expectations for the user about the differences between early stopping and LR reduction policies.

**Comment by tgaddair** [comment]
Going to close this issue for now, but @tboo please let us know if you're still seeing issues, happy to revisit.

**Comment by tboo** [comment]
Hi @w4nderlust and @tgaddair ,

thanks so much for the feedback. And sorry for getting back on that topic so late. I now have a bit a better understanding what's happening. My training behaves like that:
<img width="1081" alt="learning_curves_class_loss" src="https://user-images.githubusercontent.com/19666785/222128320-5265d164-e908-4925-bc0f-4b29bbdd3314.png">

From what I understand the learning rate is not reduced because the training loss keeps decreasing. The early stop however kicks in to avoid over-fitting based on the validation loss behavior.

Btw it would be nice if `visualize` also had some flags for manually setting axis limits. My validation loss bounces around quite heavily, which makes the auto scaled plots `ludwig visualize` provides hard to use and interpret. Below is the same data but plotted with `visualize`:

![learning_curves_class_loss](https://user-images.githubusercontent.com/19666785/222129058-28e3a5b4-4d89-49fe-979a-5d00b0943d5a.png)

Regarding MNIST: Sorry I was not aware that  `--dataset` can also read parquet files for annotations. And its a bit confusing because the user guide [refers ](https://ludwig.ai/latest/examples/mnist/#download-the-mnist-dataset) to a csv file for annotations.

Thanks again for taking the time.
